ansible/pbTestScripts/vmDestroy.sh

```diff
@@ -1,50 +1,82 @@
 #!/bin/bash
 set -eu
 
+osToDestroy=''
+force=False
 # Takes in all arguments
 processArgs()
 {
-	if [ ! -n "${WORKSPACE:-}" ]; then
-		echo "WORKSPACE not found, setting it as environment variable 'HOME'"
-		WORKSPACE=$HOME
-	fi
-	if [ $# -lt 1 ]; then
-		echo "Script takes 1 input argument: "
-		echo "The project whose VMs are to be destroyed"
-		exit 1
-	fi
-	
+	while [[ $# -gt 0 ]] && [[ ."$1" = .-* ]] ; do
+		local opt="$1";
+		shift;
+		case "$opt" in
+			"--OS" | "-o" )
+				osToDestroy=$1; shift;;
+			"--force" | "-f" )
+				force=True;;
+			"--help" | "-h" )
+				usage; exit 0;;
+			*) echo >&2 "Invalid option: ${opt}"; echo "This option was unrecognised."; usage; exit 1;;
+		esac
+	done
 }
 
-# Takes project name as arg 1, and OS as arg 2
-destroyVM()
-{
-	cd $WORKSPACE/adoptopenjdkPBTests/$1/ansible
-	ln -sf Vagrantfile.$2 Vagrantfile	# Correct Vagrantfile alias
-	vagrant destroy -f			# Force destroy without question
+usage() {
+
+		echo "Usage: ./vmDestroy <option>
+		--force | -f		Force destroy the VMs without asking confirmation
+		--help | -h		Displays this help message"
+}
+
+checkOS() {
+	local OS=$osToDestroy
+        case "$OS" in
+                "Ubuntu1604" | "U16" | "u16" )
+			osToDestroy="U16";;
+                "Ubuntu1804" | "U18" | "u18" )
+                        osToDestroy="U18";;
+                "CentOS6" | "centos6" | "C6" | "c6" )
+                        osToDestroy="C6" ;;
+                "CentOS7" | "centos7" | "C7" | "c7" )
+                        osToDestroy="C7" ;;
+                "Windows2012" | "Win2012" | "W12" | "w12" )
+                        osToDestroy="W2012";;
+                "all" )
+                        osToDestroy="U16 U18 C6 C7 W2012" ;;
+                *) echo "$OS is not a currently supported OS" ; listOS; exit 1;
```

Comment:
The error message when you get if you don't supply a parameter is not helpful as `$OS` is blank so this is confusing ... And also doesn't give any indication  that you need `-o` in front of any name from the list.